src/PerspectivePanel.test.tsx

```diff
@@ -18,43 +18,43 @@ jest.mock('@grafana/runtime', () => ({
 }));
 
 class PerspectivePanel extends PerspectivePanelOriginal {
-  componentDidMount() {
-    this.viewer.current = mockedViewer;
+  async componentDidMount() {
+    (this.viewer as any).current = mockedViewer;
     super.componentDidMount();
   }
 }
 
 const mockedViewer = {
   addEventListener: jest.fn(),
   delete: jest.fn().mockResolvedValue(true),
-  load: jest.fn().mockResolvedValue(),
+  load: jest.fn().mockResolvedValue({}),
   notifyResize: jest.fn(),
   save: jest.fn().mockReturnValue({}),
-  toggleConfig: jest.fn().mockResolvedValue(),
+  toggleConfig: jest.fn().mockResolvedValue(true),
 };
 
 const mockedHeight = 100;
 const mockedWidth = 200;
 
 const mockedData: PanelData = {
+  state: LoadingState.Done,
   series: [
     toDataFrame({ refId: 'AA', fields: [{ name: 'AA', values: [0, 1, 2, 3] }] }),
     toDataFrame({ refId: 'BB', fields: [{ name: 'BB', values: [4, 5, 6, 7] }] }),
   ],
-};
+} as PanelData;
 
-let mockedIsDark;
+let mockedIsDark = false;
 
-const createPanel = config => {
-  const { data, height, options, width } = {
+const createPanel = (config?: any) => {
+  const props = {
+    options: {}, // default empty
+    ...config,
```

Comment:
Putting this first means that the defaults cannot be overridden.

Reply:
doesn't it only override options?

I think this does the same thing it used to...  I am not trying to change any behavior, just getting it to compile

Reply:
If there's a `config.height`, it'll be ignored unless the `...config` comes after